packages/calcite-components/src/components/block-group/block-group.tsx

```diff
@@ -0,0 +1,412 @@
+// @ts-strict-ignore
+import Sortable from "sortablejs";
+import { debounce } from "lodash-es";
+import { PropertyValues } from "lit";
+import { LitElement, property, createEvent, h, method, state, JsxNode } from "@arcgis/lumina";
+import {
+  InteractiveComponent,
+  InteractiveContainer,
+  updateHostInteraction,
+} from "../../utils/interactive";
+import { createObserver } from "../../utils/observers";
+import {
+  connectSortableComponent,
+  disconnectSortableComponent,
+  SortableComponent,
+} from "../../utils/sortableComponent";
+import {
+  componentFocusable,
+  LoadableComponent,
+  setComponentLoaded,
+  setUpLoadableComponent,
+} from "../../utils/loadable";
+import { MoveEventDetail, MoveTo, ReorderEventDetail } from "../sort-handle/interfaces";
+import { DEBOUNCE } from "../../utils/resources";
+import { Block } from "../block/block";
+import { focusFirstTabbable, getRootNode } from "../../utils/dom";
+import { guid } from "../../utils/guid";
+import { blockGroupSelector, blockSelector, CSS } from "./resources";
+import { styles } from "./block-group.scss";
+import { BlockDragDetail } from "./interfaces";
+import { updateBlockChildren } from "./utils";
+
+declare global {
+  interface DeclareElements {
+    "calcite-block-group": BlockGroup;
+  }
+}
+
+/**
+ * @slot - A slot for adding `calcite-block` elements.
+ */
+export class BlockGroup
+  extends LitElement
+  implements InteractiveComponent, LoadableComponent, SortableComponent
+{
+  // #region Static Members
+
+  static override styles = styles;
+
+  // #endregion
+
+  // #region Private Properties
+
+  dragSelector = blockSelector;
+
+  handleSelector = "calcite-sort-handle";
+
+  private items: Block["el"][] = [];
+
+  mutationObserver = createObserver("mutation", () => {
+    this.updateBlockItems();
+  });
+
+  private parentBlockGroupEl: BlockGroup["el"];
+
+  sortable: Sortable;
+
+  private updateBlockItems = debounce((): void => {
+    this.updateGroupItems();
+    const { dragEnabled, el, moveToItems } = this;
+
+    const items = Array.from(this.el.querySelectorAll(blockSelector));
+
+    items.forEach((item) => {
+      if (item.closest(blockGroupSelector) === el) {
+        item.moveToItems = moveToItems.filter(
+          (moveToItem) => moveToItem.element !== el && !item.contains(moveToItem.element),
+        );
+        item.dragHandle = dragEnabled;
+      }
+    });
+
+    if (!this.parentBlockGroupEl) {
+      this.items = items;
+    }
+
+    this.setUpSorting();
+  }, DEBOUNCE.nextTick);
+
+  // #endregion
+
+  // #region State Properties
+
+  @state() assistiveText: string;
+
+  @state() moveToItems: MoveTo[] = [];
+
+  // #endregion
+
+  // #region Public Properties
+
+  /** When provided, the method will be called to determine whether the element can  move from the component. */
```

Comment:
nitpick:
- Remove extra space
- Change move -> sort for more clarity

```suggestion
  /** When provided, the method will be called to determine whether the element is sortable in the component. */
```